systemtests/src/main/java/io/enmasse/systemtest/scale/MessagingPerformanceTestManager.java

```diff
@@ -0,0 +1,150 @@
+import io.enmasse.systemtest.logs.CustomLogger;
+import io.enmasse.systemtest.model.address.AddressType;
+import io.enmasse.systemtest.platform.Kubernetes;
+import io.enmasse.systemtest.platform.apps.SystemtestsKubernetesApps;
+import io.enmasse.systemtest.scale.metrics.MessagingClientMetricsClient;
+import io.enmasse.systemtest.scale.metrics.MetricsMonitoringResult;
+import io.enmasse.systemtest.utils.TestUtils;
+
+/**
+ * This class should be instantiated once per test
+ */
+public class MessagingPerformanceTestManager {
+
+    private final Logger logger = CustomLogger.getLogger();
+
+    private final Kubernetes kubernetes;
+
+    private Supplier<ScaleTestClientConfiguration> clientProvider;
+
+    private int totalConnections = 0;
+    private final Map<String, ScaleTestClient<MessagingClientMetricsClient>> clientsMap = new ConcurrentHashMap<>();
+    private final Queue<String> clientsMonitoringQueue = new ConcurrentLinkedQueue<>();
+    private final AtomicReference<MetricsMonitoringResult> monitoringResult = new AtomicReference<MetricsMonitoringResult>(new MetricsMonitoringResult());
+
+    public MessagingPerformanceTestManager(Endpoint addressSpaceEndpoint, UserCredentials credentials) {
+        this.clientProvider = () -> {
+            ScaleTestClientConfiguration client = new ScaleTestClientConfiguration();
+            client.setHostname(addressSpaceEndpoint.getHost());
+            client.setPort(addressSpaceEndpoint.getPort());
+            client.setUsername(credentials.getUsername());
+            client.setPassword(credentials.getPassword());
+            return client;
+        };
+        this.kubernetes = Kubernetes.getInstance();
+    }
+
+    public int getConnections() {
+        return totalConnections;
+    }
+
+    public int getClients() {
+        return clientsMap.size();
+    }
+
+    public AtomicReference<MetricsMonitoringResult> getMonitoringResult() {
+        return monitoringResult;
+    }
+
+    public void deployClient(List<Address> addresses, AddressType type, int linksPerConnection) throws Exception {
+        if (addresses == null || addresses.isEmpty()) {
+            throw new IllegalArgumentException("Addresses cannot be null or empty");
+        }
+        var addressesOfType = addresses.stream()
+                .filter(a -> a.getSpec().getType().equals(type.toString()))
+                .map(a -> a.getSpec().getAddress())
+                .toArray(String[]::new);
+        var clientConfig = clientProvider.get();
+        clientConfig.setClientType(ScaleTestClientType.messaging);
+        clientConfig.setAddressesType(type);
+        clientConfig.setAddresses(addressesOfType);
+        clientConfig.setLinksPerConnection(linksPerConnection);
+
+        SystemtestsKubernetesApps.deployScaleTestClient(kubernetes, clientConfig);
+
+        int connectionsInThisClient = (addressesOfType.length/linksPerConnection) * 2; // *2 because client creates sender and receiver
```

Comment:
We should make this configurable, i.e. for the IoT case where you might have a lot more senders than receivers.

Reply:
that will require to implement some login in the client, but it's ok and I like it, so I'll add it to my TODO list